src/main/java/Main.java

```diff
@@ -1,29 +1,35 @@
 import org.apache.commons.codec.binary.Base64InputStream;
 
 import java.io.*;
+import java.util.Properties;
 import java.util.zip.GZIPInputStream;
 
 public class Main {
     public static void main(String[] args) {
-        File file = new File(Main.class.getResource("file.txt.gz").getFile());
-        byte[] buffer = new byte[1024];
-        int signal;
-        String decodedString = "";
         try {
-            InputStream fileInputStream = new FileInputStream(file);
-            GZIPInputStream gzipInputStream = new GZIPInputStream(fileInputStream);
-            Base64InputStream base64InputStream = new Base64InputStream(gzipInputStream);
-            signal = base64InputStream.read(buffer);
-            while (signal != -1) {
-                decodedString += new String(buffer);
-                signal = base64InputStream.read(buffer);
-            }
-            gzipInputStream.close();
-        } catch (FileNotFoundException e) {
-            e.printStackTrace();
+            writeOutput();
         } catch (IOException e) {
             e.printStackTrace();
         }
-        System.out.println(decodedString);
+    }
+
+    private static Properties readProperties() throws IOException {
+        PropertiesReader propertiesReader = new PropertiesReader();
+        return propertiesReader.read("conf.properties");
+    }
+
+    private static InputStreamReader createInputStreamPath(Properties properties) throws IOException {
```

Comment:
do you need the entire `Properties` object or just the input file's path? if later down the line you will need to change it so that the properties are received through a CLI or a using a REST API, will this method still require a `Properties` object?